### PR TITLE
ObjectConstructor considers context serialization groups for identifiers

### PR DESCRIPTION
**Pull request by bertuz**
If a property set as identifier (@ORM\Id) does not belong to the defined group context, and the object to be deserialized has it, then use the fallback object constructor

### CODE REVIEW

**Comment by goetas** [comment]
Thanks for the pull request.

Unfortunately is really had for me to understand the usecase for this. 
Can you create a test case for it? (Anyway without proper tests can not merge it)

File: src/JMS/Serializer/Construction/DoctrineObjectConstructor.php [hunk 1]
```diff
@@ -43,6 +47,30 @@ public function __construct(ManagerRegistry $managerRegistry, ObjectConstructorI
...
     }
 
+    public function setAnnotationDriver(AnnotationDriver $annotationDriver)
+    {
+        $this->annotationDriver = $annotationDriver;
+    }
+
+    protected function getPropertyGroups(JMSClassMetadata $metadata, $property)
+    {
+        $classMetadata = $this->annotationDriver->loadMetadataForClass($metadata->reflection);
```
Comment (goetas) [inline src/JMS/Serializer/Construction/DoctrineObjectConstructor.php]: What happens if nobody calls the setAnnotationDriver

File: src/JMS/Serializer/Construction/DoctrineObjectConstructor.php [hunk 2]
```diff
@@ -43,6 +47,30 @@ public function __construct(ManagerRegistry $managerRegistry, ObjectConstructorI
...
+
+        // up to the hierarchy
+        while ($classMetadata) {
+            // limit case
+            if (array_key_exists($property, $classMetadata->propertyMetadata)) {
+                $propertyGroups = $classMetadata->propertyMetadata[$property]->groups ?? array();
```
Comment (goetas) [inline src/JMS/Serializer/Construction/DoctrineObjectConstructor.php]: ?? Is only for php 7

**Comment by goetas** [comment]
Tests are green because your code is never executed

**Comment by bertuz** [comment]
Hi @goetas, I'm gonna close this PR since I had to start over. I've opened a new PR with a better explanation, a proper test coverage and the fix I wanted to pull-request https://github.com/schmittjoh/serializer/pull/812

**Pull request closed without merging**